Storage/ThirdParty/SwiftData.swift

```diff
@@ -157,22 +194,28 @@ open class SwiftData {
             deferred.running = true
             defer {
                 deferred.running = false
+                deferred.connection = nil
             }
 
-            if !self.closed && self.sharedConnection == nil {
-                self.sharedConnection = ConcreteSQLiteDBConnection(filename: self.filename, flags: SwiftData.Flags.readWriteCreate.toSQL(), key: self.key, prevKey: self.prevKey, schema: self.schema, files: self.files)
+            if !self.closed {
```

Comment:
no early return here if the db is closed?

Reply:
No, because if it is closed, we need to return a `FailedSQLiteDBConnection`.